### PR TITLE
Fix borg default names being assigned as custom names (#61080)

### LINKED ISSUE #61080: Borg names don't change on model change anymore
## Round ID:
Over multiple rounds

## Reproduction:
Previously when a borg chose a model it would go from being "Default Cyborg-985" to something like "Engineering Cyborg-985", but this stopped happening at some point.

### PR DESCRIPTION
**Pull request by Prometheos2**
## About The Pull Request

The PR focuses on bugs about silicon robots' names (borgs & shells): the default names and a potential logic error on renaming boards. 

This is my first PR on this repository, so I would be glad to have feedback if there is something wrong or too unusual. I would also like feedback on language: whether this is too verbose, or gibberish.

I draft it as I am not finished with the initial tests yet, but I would be glad to receive feedback and make changes as I complete the tests.

#### Tests to do:

- [x] Borgs' default names are changed w/ model
- [ ] Shells' default names are changed w/ model
- [ ] Borgs' custom names are properly used on init
- [ ] Borgs' custom names are *not* changed w/ model
- [ ] AI is notified of actual renames
- [ ] AI is not notified of 'false' renames

### Default names (#61080)
Fixes  #61080

First, default names—_Default Cyborg-[id]_. 
Previously, they used to change along with the model, resulting in _Engineering Cyborg_, _Medical Cyborg_, etc. This PR fix those problems. [tested for borgs; test pending for shells]

You may find a similar mechanic with shells: after changing the model, leaving and deploying again, the name reflects its model (names reset on deployment). However, shells should benefit from this bugfix as well, and be renamed on model change.

I removed the custom name assignation from the overridden *fully_replace_character_name* proc, to avoid this issue. However, I think that my implementation might be considered messy, and I would like to know if it would be better to override the *apply_pref_name* proc to add it there.

### Renaming boards [test pending]
I noticed a potential logic error—which I have yet to test to make sure it *is* an error and not just me.
Apparently, if the previous and new names for a borg are the same, the AI is notified of the rename of borg 'X' to borg 'X'; instead of notifying actual renames.

If you think it should be in another PR, feel free to tell me. I'm afraid this might be a bit of a scope creep for this PR.

#### Side notes
While it is not addressed in the changes, I noticed that shells bypass the condition to get a custom name on deployment, so using renaming boards would only rename it for a given deployment and 'waste' it on next deployment.

## Why It's Good For The Game

This PR fixes a bug I was personally bothered with, as well as #61080's submitter.
Additionally, it fixes a potential logic error that surely bothered a few AI players, if it *is* an error.

## Changelog

:cl: Prometheos2
fix: properly update default borg names on model change
fix: warn the AI of borg rename by boards instead of canceled renames.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
This reminds me, I don't think it's possible to set your borg name to be the default with the new preferences menu.

**Comment by Prometheos2** [comment]
> This reminds me, I don't think it's possible to set your borg name to be the default with the new preferences menu.

You're right, thanks for pointing that out.
I will try to see if I could fix it, although I doubt I have what it takes yet 😥

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -43,14 +43,25 @@
...
+ * Renames cyborgs when clicked on with a renaming board.
+ *
+ * Updates the cyborg's real_name, key_name, and custom_name to the choosen name (heldname).
+ * Notify the AI if there is a change.
+ * Note: Shells reset their names on deployment, so renaming them is not as useful as it is for cyborgs.
+ *
+ * Arguments:
+ * * R — a robot: a cyborg or a shell
+ * * user — the character to use the board on the cyborg
+ */
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_upgrades.dm:56]: we dont do documentation on procs that already exist i think

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -656,14 +657,24 @@
...
+ * Fully update the name of a cyborg/shell.
+ *
+ * Overrides and calls [/mob/proc/fully_replace_character_name].
+ * Updates the borg's name, real_name, and its camera's identification tag.
+ * It is called on cyborg initialization, model changes and renames (i.e., renaming board)
+ *
+ * Arguments:
+ * * oldname — The to-be-replaced real_name
+ * * newname — The new name
+ */
```
Comment (Fikou) [inline code/modules/mob/living/silicon/robot/robot.dm:670]: same

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 1 above)
Comment (Prometheos2) [reply code/game/objects/items/robot/robot_upgrades.dm]: Alright, I removed them
Thanks for pointing it out

**Comment by Prometheos2** [comment]
I'm busy with deadlines at university; I will open the PR once I've time to properly run the tests.

**Comment by LemonInTheDark** [comment]
Man you test more then half our coders, look forward to what you do when you're done. 
(If this is in fact my fault thanks for cleaning up my mess)

**Comment by Prometheos2** [comment]
> Man you test more then half our coders
It's mostly to reduce the number of bugs I create; I would hate to see a hotfix because of my changes.
Also, because I'm not sure if I'm reading the rename announcement's logic correctly

I have yet to find a proper balance between too much test and too few 😄
I wonder if I could do some of those test through unit testing, though.

>look forward to what you do when you're done.
Glad to hear it
Sadly, I'm struggling a bit with university; I have to keep myself from booting up DreamDaemon 😥

>(If this is in fact my fault thanks for cleaning up my mess)
I don't want to point fingers, but I don't think it was not your changes that induced the bugs.

**Pull request closed without merging**